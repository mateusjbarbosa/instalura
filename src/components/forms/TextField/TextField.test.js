import React from 'react';

import user from '@testing-library/user-event';

import { render, screen } from '../../../infra/test/testUtils';

import TextField from '.';

describe('<TextField />', () => {
  test('renders component', () => {
    render(
      <TextField
        placeholder="Nome"
        value="test"
        onChange={() => {}}
        name="name"
      />
      ,
    );

    const textField = screen.getByPlaceholderText(/nome/i);

    expect(textField).toMatchSnapshot();
  });

  describe('when field is valid', () => {
    describe('and user is typing', () => {
      test('the value must be updated', () => {
        const onChangeMock = jest.fn();

        render(
          <TextField
            placeholder="E-mail"
            value=""
            onChange={onChangeMock}
            name="email"
            isTouched
          />
          ,
        );

        const inputEmail = screen.getByPlaceholderText(/e-mail/i);
        const emailValue = 'samemail@email.com';

        user.type(inputEmail, emailValue);

        expect(onChangeMock).toHaveBeenCalledTimes(emailValue.length);
      });
    });

    describe('when field is invalid', () => {
      test('displays the respective error message', () => {
        render(
          <TextField
            placeholder="E-mail"
            value="samemail@email.com"
            onChange={() => {}}
            name="email"
            isTouched
            error="O campo email é obrigatório"
          />
          ,
        );

        const inputEmail = screen.getByPlaceholderText(/e-mail/i);

        expect(inputEmail).toHaveValue('samemail@email.com');
        expect(screen.getByRole('alert')).toHaveTextContent('O campo email é obrigatório');
        expect(inputEmail).toMatchSnapshot();
      });
    });
  });
});
