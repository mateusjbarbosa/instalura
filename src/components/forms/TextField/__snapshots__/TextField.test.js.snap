// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders component 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

.c1 {
  width: 100%;
  border: 1px solid #88989E;
  padding: 12px 16px;
  outline: 0;
  border-radius: 12px;
}

<input
  class="c0 c1"
  name="name"
  placeholder="Nome"
  type="text"
  value="test"
/>
`;

exports[`<TextField /> when field is valid when field is invalid displays the respective error message 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

.c1 {
  width: 100%;
  border: 1px solid #88989E;
  padding: 12px 16px;
  outline: 0;
  border-radius: 12px;
}

<input
  class="c0 c1"
  name="email"
  placeholder="E-mail"
  type="text"
  value="samemail@email.com"
/>
`;
